### PR TITLE
Limit the total size of uploaded files to 1MB

### PR DESCRIPTION
**Pull request by nachtm**
This sets an arbitrary limit of 1MB on the size of files that a user can upload. If a user uploads a set of files larger than 1MB, we pop up a dialog that requests that the user pick a different set of files. 

The UX here is a little weird because the user has no indication of what files are "on deck" when they hit upload. Something to think about. I also refactored the dialogs a bit since there was a lot of shared code between the various popups on the form -- now, we have a generic FormInfoDialog component that we can populate with a variety of information. 

This is what the popup looks like:
image

### CODE REVIEW

**Comment by danielduhh** [comment]
Any reason why you chose 1MB? This is pretty small given folks will be uploading high quality photos from their smart phones
@nachtm

**Comment by nachtm** [comment]
@danielduhh Not really -- I just looked at the pricing and it's $.0026 per GB, so not too bad.

What do you think the limit should be?

**Comment by danielduhh** [comment]
I uploaded a couple photos from my iphone and they ranged from 670kb to 1.6mb. Let's do some more testing but I think 5MB should be safe for now..

Are you planning on wiring up google analytics?? We should keep track of the size of each upload, as well as errors like `LARGE_FILES`.

**Comment by nachtm** [comment]
@danielduhh That wasn't in the roadmap for the initial release, but we could add it on as a nice-to-have for our initial release/a to-do for a second cycle. Unless you think it's important to have analytics on release day!

File: src/components/Form.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ import LoadingOverlay from 'react-loading-overlay';
 import MediaUpload from './MediaUpload';
 import FormMap from './FormMap';
 import NeighborhoodService from '../services/NeighborhoodService';
+import FormInfoDialog from './FormInfoDialog';
```
Comment (sayas01) [inline src/components/Form.js]: awesome! 
nit: remove unused import statements for dialog modules

File: src/components/Form.js [hunk 2]
```diff
@@ -40,7 +41,16 @@ const counts = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
 // constants
 const THANKS_FOR_SUBMITTING = 'Thank you for your submission! Please note that the system will display your observation on the map after a period of one week.';
 const ERROR_ON_SUBMISSION = 'Something went wrong during your submission. Please try again later.';
+const FILES_TOO_LARGE = 'Please choose a set of files to upload that are smaller than 5MB in total.';
+const MAX_FILE_SIZE = 5242880; // 5MiB
 const neighborhoodService = new NeighborhoodService();
+const DIALOG_MODES = {
```
Comment (sayas01) [inline src/components/Form.js]: neat!

**Pull request merged**